streams/shared/src/main/scala/scalaz/stream/ZStream.scala

```diff
@@ -660,6 +660,38 @@ trait ZStream[-R, +E, +A] extends Serializable { self =>
         }
       }
   }
+
+  /**
+   * Stores this stream to a fixed bounded queue and streams data from the bounded queue.
+   */
+  final def buffer[R1 <: R, E1 >: E](capacity: Int = 1): ZStream[R1, E1, A] =
+    new ZStream[R1, E1, A] {
+      override def fold[R2 <: R1, E2 >: E1, A1 >: A, S]: Fold[R2, E2, A, S] =
```

Comment:
Would it be possible to reuse the implementations of `toQueue` and `fromQueue` here?

Reply:
Fixed